libenkf/src/enkf_config_node.c

```diff
@@ -565,6 +566,9 @@ void enkf_config_node_free(enkf_config_node_type * node) {
   free(node->key);
   stringlist_free(node->obs_keys);
 
+  if (node->init_file_abs_path != NULL)
```

Comment:
The code already has many of these checks - but they are actually not necessary. Calling `free( NULL )` is a well defined no-op: http://stackoverflow.com/questions/1938735/does-freeptr-where-ptr-is-null-corrupt-memory